spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncher.java

```diff
@@ -78,6 +79,13 @@ public LocalTaskLauncher(LocalDeployerProperties properties) {
 	@Override
 	public String launch(AppDeploymentRequest request) {
 
+		if (this.maxConcurrentExecutionsReached()) {
+			throw new IllegalStateException(
+				String.format("Cannot launch task %s. The maximum concurrent task executions is at its limit [%d].",
+					request.getDefinition().getName(), this.getLocalDeployerProperties().getMaximumConcurrentTasks())
+			);
+		}
```

Comment:
While I'm not advocating for anything other than this now, we will need to consider how to make this more robust in the future (queuing mechanism, etc).